data-prepper-plugins/dynamodb-source-coordination-store/src/main/java/org/opensearch/dataprepper/plugins/sourcecoordinator/dynamodb/DynamoDbSourceCoordinationStore.java

```diff
@@ -23,33 +32,102 @@
     private final PluginMetrics pluginMetrics;
+    private final DynamoDbClientWrapper dynamoDbClientWrapper;
 
     @DataPrepperPluginConstructor
     public DynamoDbSourceCoordinationStore(final DynamoStoreSettings dynamoStoreSettings, final PluginMetrics pluginMetrics) {
         this.dynamoStoreSettings = dynamoStoreSettings;
         this.pluginMetrics = pluginMetrics;
+        this.dynamoDbClientWrapper = DynamoDbClientWrapper.create(dynamoStoreSettings.getRegion(), dynamoStoreSettings.getStsRoleArn());
     }
+
+    @Override
+    public void initializeStore() {
+        dynamoDbClientWrapper.tryCreateTable(dynamoStoreSettings.getTableName(), constructProvisionedThroughput(
+                dynamoStoreSettings.getProvisionedReadCapacityUnits(), dynamoStoreSettings.getProvisionedWriteCapacityUnits()));
+    }
+
     @Override
     public Optional<SourcePartitionStoreItem> getSourcePartitionItem(final String partitionKey) {
-        return Optional.empty();
+        return dynamoDbClientWrapper.getSourcePartitionItem(partitionKey);
     }
 
     @Override
     public boolean tryCreatePartitionItem(final String partitionKey,
                                           final SourcePartitionStatus sourcePartitionStatus,
                                           final Long closedCount,
                                           final String partitionProgressState) {
-        return false;
+        final DynamoDbSourcePartitionItem newPartitionItem = new DynamoDbSourcePartitionItem();
+        newPartitionItem.setSourcePartitionKey(partitionKey);
+        newPartitionItem.setSourcePartitionStatus(sourcePartitionStatus);
+        newPartitionItem.setClosedCount(closedCount);
+        newPartitionItem.setPartitionProgressState(partitionProgressState);
+        newPartitionItem.setVersion(0L);
+
+        return dynamoDbClientWrapper.tryCreatePartitionItem(newPartitionItem);
     }
 
     @Override
-    public Optional<SourcePartitionStoreItem> tryAcquireAvailablePartition() {
+    public Optional<SourcePartitionStoreItem> tryAcquireAvailablePartition(final String ownerId, final Duration ownershipTimeout) {
+        final Optional<PageIterable<DynamoDbSourcePartitionItem>> dynamoDbSourcePartitionItemPageIterable =
+                dynamoDbClientWrapper.getSourcePartitionItems(Expression.builder()
+                        .expressionValues(Map.of(
+                                ":unassigned", AttributeValue.builder().s(SourcePartitionStatus.UNASSIGNED.name()).build(),
+                                ":assigned", AttributeValue.builder().s(SourcePartitionStatus.ASSIGNED.name()).build(),
+                                ":closed", AttributeValue.builder().s(SourcePartitionStatus.CLOSED.name()).build(),
+                                ":t", AttributeValue.builder().s(Instant.now().toString()).build(),
+                                ":ro", AttributeValue.builder().s(Instant.now().toString()).build(),
```

Comment:
I missed this in the first pass - let's avoid abbreviations for the field names

Reply:
Yeah I thought there was a restriction on that from ddb but turns out that is not the case